### PR TITLE
Add pundit policies for Verification documents controller

### PR DESCRIPTION
**Pull request by saipraveen18**
# PR Checklist

Please check if your PR fulfills the following requirements
- [ ] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [X] Tests for the changes have been added (for bugfixes/features)

# PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/n/projects/2640059/stories/187136947#

# A brief description of the changes

Current behavior: Verification documents controller do not have any pundit policies

New behavior: Add pundit policies for verification documents controller actions

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/controllers/insured/verification_documents_controller.rb [hunk 1]
```diff
@@ -47,19 +54,18 @@ def download
...
-    @family = @person.primary_family
+  def check_for_consumer_role
+    @consumer_role = @person.consumer_role
+    return if @consumer_role
+
+    flash[:error] = "No consumer role exists, you are not authorized to upload documents"
+    redirect_to verification_insured_families_path
   end
 
   def person_consumer_role
```
Comment (saikumar9) [inline app/controllers/insured/verification_documents_controller.rb:71]: @saipraveen18 We do not need this method anymore.

File: app/controllers/insured/verification_documents_controller.rb [hunk 2]
```diff
@@ -3,10 +3,14 @@
...
 
-  before_action :get_family
-  before_action :updateable?, :find_type, :find_docs_owner, only: [:upload]
+  before_action :set_current_person
+  before_action :find_type, :find_docs_owner, only: [:upload]
+  before_action :check_for_consumer_role
 
   def upload
+    consumer_role = person_consumer_role
+    authorize consumer_role, :verification_document_upload?
```
Comment (saikumar9) [inline app/controllers/insured/verification_documents_controller.rb:12]: @saipraveen18 You are already assigning `@person.consumer_role` to `@consumer_role`. You can use it directly like the one below.

```suggestion
    authorize @consumer_role, :verification_document_upload?
```

File: app/controllers/insured/verification_documents_controller.rb [hunk 3]
```diff
@@ -34,6 +38,9 @@ def upload
   end
 
   def download
+    consumer_role = person_consumer_role
+    authorize consumer_role, :verification_document_download?
```
Comment (saikumar9) [inline app/controllers/insured/verification_documents_controller.rb:42]: @saipraveen18 You are already assigning `@person.consumer_role` to `@consumer_role`. You can use it directly like the one below.

```suggestion
    authorize @consumer_role, :verification_document_upload?
```

**Review by saikumar9 — CHANGES_REQUESTED** [review]

File: app/controllers/insured/verification_documents_controller.rb (hunk 3 above)
Comment (saipraveen18) [reply app/controllers/insured/verification_documents_controller.rb]: Updated

**Review by nisanthyaganti9 — APPROVED** [review]

**Comment by nisanthyaganti9** [comment]
LGTM, branch is red right now, but this seems like a flickering one.

**Review by saikumar9 — APPROVED** [review]
lgtm!

**Pull request merged**